pyiron/atomistics/structure/neighbors.py

```diff
@@ -695,7 +716,9 @@ def find_neighbors_by_vector(self, vector, deviation=False):
             return indices[np.arange(len(dist)), np.argmin(dist, axis=-1)], np.min(dist, axis=-1)
         return indices[np.arange(len(dist)), np.argmin(dist, axis=-1)]
 
-    def cluster_by_vecs(self, bandwidth=None, n_jobs=None, max_iter=300):
+    def cluster_by_vecs(
+        self, distance_threshold=None, n_clusters=None, linkage='complete', affinity='euclidean'
+    ):
```

Comment:
Maybe we want to group the clustering specific arguments into a dictionary like scipy functions do sometimes [see here under options](https://docs.scipy.org/doc/scipy/reference/generated/scipy.optimize.minimize.html#scipy.optimize.minimize).  Not necessary if you're convinced that this clustering method will stay for a while.

Reply:
Do we ever want to change the metric though?

Reply:
> Maybe we want to group the clustering specific arguments into a dictionary like scipy functions do sometimes [see here under options](https://docs.scipy.org/doc/scipy/reference/generated/scipy.optimize.minimize.html#scipy.optimize.minimize). Not necessary if you're convinced that this clustering method will stay for a while.

That's actually a really good idea. Probably the best is to use `AgglomerativeClustering` for systems with a small number of atoms and `DBSCAN` with larger systems. I don't know exactly what would be the best way though. Do you perhaps have a specific idea?

Reply:
> Do we ever want to change the metric though?

One thing is by using the Manhattan norm (or L1 norm), you can potentially enhance the angular variation, although that's orientation dependent, so I wouldn't generalise my claim here. Another thing is, if for some reason the items are periodic, you might want to put a norm with periodicity, even though I saw that [it makes the algorithm extremely slow](https://stackoverflow.com/questions/32938494/dbscan-with-metric-only-in-scikit-learn). So, to answer your question: No, probably not, but it's not impossible that we see such a case.

Reply:
Will open a separate PR for that.

Reply:
I'm personally very much in favour of either user being able to switch between algorithms or pyiron automatically detecting which one is better. The one I implemented here is not much more than a mere replacement of `MeanShift` by `AgglomerativeClustering`.